src/Kovan-DssSpell.t.sol

```diff
@@ -1278,71 +1672,116 @@ contract DssSpellTest is DSTest, DSMath {
-                hevm.warp(castTime - 1 seconds); // 13:59:59 UTC
-                assertEq(spell.nextCastTime(), castTime);
+        // Add balance to the test address
+        uint256 ilkAmt = 1 * THOUSAND * WAD;
 
-                hevm.warp(castTime + 7 hours + 1 seconds); // 21:00:01 UTC
-                if (i < 4) {
-                    assertEq(spell.nextCastTime(), monday_1400_UTC + (i + 1) * 1 days); // Next day at 14:00 UTC
-                } else {
-                    assertEq(spell.nextCastTime(), monday_1400_UTC + 7 days); // Next monday at 14:00 UTC (friday case)
-                }
-            }
-        }
-    }
+        hevm.store(
+            address(LINK),
+            keccak256(abi.encode(address(this), uint256(0))),
+            bytes32(ilkAmt)
+        );
 
-    function testFail_notScheduled() public view {
-        spell.nextCastTime();
-    }
+        assertEq(LINK.balanceOf(address(this)), ilkAmt);
 
-    function test_use_eta() public {
-        hevm.warp(1606161600); // Nov 23, 20 UTC (could be cast Nov 26)
+        // Contracts set
+        assertEq(dog.vat(), address(vat));
+        assertEq(dog.vow(), address(vow));
+        (address clip,,,) = dog.ilks("LINK-A");
+        assertEq(clip, address(clipLINK));
+        assertEq(clipLINK.ilk(), "LINK-A");
+        assertEq(clipLINK.vat(), address(vat));
+        assertEq(clipLINK.vow(), address(vow));
+        assertEq(clipLINK.dog(), address(dog));
+        assertEq(clipLINK.spotter(), address(spotter));
+        assertEq(clipLINK.calc(), addr.addr("MCD_CLIP_CALC_LINK_A"));
 
-        vote(address(spell));
-        spell.schedule();
+        // Authorization
+        assertEq(flipLINK.wards(address(cat))    , 0);
+        assertEq(flipLINK.wards(address(flipMom)), 0);
+
+        assertEq(vat.wards(address(clipLINK))    , 1);
+        assertEq(dog.wards(address(clipLINK))    , 1);
+        assertEq(clipLINK.wards(address(dog))    , 1);
+        assertEq(clipLINK.wards(address(end))    , 1);
+        assertEq(clipLINK.wards(address(clipMom)), 1);
+
+        // Join to adapter
+        assertEq(vat.gem("LINK-A", address(this)), 0);
+        LINK.approve(address(joinLINK), ilkAmt);
+        joinLINK.join(address(this), ilkAmt);
+        assertEq(LINK.balanceOf(address(this)), 0);
+        assertEq(vat.gem("LINK-A", address(this)), ilkAmt);
 
-        uint256 castTime = spell.nextCastTime();
-        assertEq(castTime, spell.eta());
+        // Generate new DAI to force a liquidation
+        LINK.approve(address(joinLINK), ilkAmt);
+        joinLINK.join(address(this), ilkAmt);
+        (,,uint256 spot,,) = vat.ilks("LINK-A");
+        // dart max amount of DAI
+        vat.frob("LINK-A", address(this), address(this), address(this), int(ilkAmt), int(mul(ilkAmt, spot) / RAY));
+        hevm.warp(now + 1);
+        jug.drip("LINK-A");
+        assertEq(clipLINKA.kicks(), 0);
+        dog.bark("LINK-A", address(this), address(this));
+        assertEq(clipLINKA.kicks(), 1);
```

Comment:
Maybe extend this test to run through the whole liquidation process.

Reply:
Done!